doc/changes.rst

```diff
@@ -1,12 +1,302 @@
+  * Applicable to ``install``, ``upgrade``, and related commands.
+
+``--dump-main-config``
+  * Along with related ``--dump-repo-config=REPO_ID``, these are new options to print configuration values on the standard output.
+
+``--offline``
+  * Store the transaction to be performed offline.
+  * Applicable to all relevant transactional commands.
+  * See the :ref:`Offline command <offline_command_ref-label>` for more information.
+
+``--show-new-leaves``
+  * Show newly installed leaf packages and packages that became leaves after a transaction.
+
+``--skip-unavailable``
+  * Allow skipping packages that are not available in repositories.
+  * Not to be confused with the :ref:`skip_if_unavailable <skip_if_unavailable_options-label>` configuration option.
+  * Applicable to ``install``, ``upgrade``, and related commands.
+  * See also the :ref:`strict option deprecation <strict_option_conf_changes_ref-label>` for more information.
+
+``--use-host-config``
+  * See the :ref:`main man page <use_host_config_option_ref-label>` for more details.
+
+
+Commands
+--------
+
+Optional arguments
+^^^^^^^^^^^^^^^^^^
+Commands cannot have optional subcommands and optional arguments. Optional subcommands were ambiguous,
+making it unclear whether the input was intended as a command argument or a subcommand. Subcommands are now mandatory if present.
+
+Examples:
+  * Before: ``dnf history <transaction ID>`` Now: ``dnf history info <transaction ID>``
+  * Before: ``dnf updateinfo`` Now: ``dnf updateinfo summary``
+
+
+Changes to individual commands
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+``alias``
+  * Dropped. The command has been replaced by a different functionality.
+  * See the :ref:`Aliases for command line arguments <aliases_misc_ref-label>` for more information.
+
+``automatic``
+  * Now a DNF5 plugin.
+  * Different configuration file locations and some changes in format.
+  * See the :ref:`Automatic command <automatic_plugin_ref-label>` for more information.
+
+``autoremove``
+  * Dropped the ``<spec>`` positional argument since the use case is sufficiently covered by the ``remove`` command.
+  * Specific variants of the command, ``autoremove-n``, ``autoremove-na``, and ``autoremove-nevra``, are not supported anymore.
+
+``builddep``
+  * Dropped ``--spec`` and ``--srpm`` arguments as automatic detection from file extensions is implemented now.
+
+``config-manager``
+  * New behavior introduced.
+  * Parameters are replaced by subcommands.
+
+    * Examples:
+
+      * Before: ``--add-repo`` Now: ``addrepo``
+      * Before: ``--save --setopt`` Now: ``setopt``
+
+  * Existing repository files are not modified; drop-in override files are created instead.
+  * See the :ref:`config-manager documentation <config_manager_plugin_ref-label>` for more information.
+
+``distro-sync``
+  * Now when any argument does not match any package or is not installed, DNF5 fail. The behavior can be modified by the ``--skip-unavailable`` option.
```

Comment:
This sentence feels clunky to me and I am not sure I understand it. I think that at least it should be `..., DNF5 fails`.

What about: `DNF5 fails when an argument doesn't match any installed package.` or `Now when any argument does not match an installed package DNF5 fails.`?

It is repeated several times.